iff --git a/make/devkit/Tools.gmk b/make/devkit/Tools.gmk
index 249eaa66247..56e348b29b6 100644
--- a/make/devkit/Tools.gmk
+++ b/make/devkit/Tools.gmk
@@ -68,7 +68,8 @@ else ifeq ($(BASE_OS), Fedora)
     BASE_OS_VERSION := $(DEFAULT_OS_VERSION)
   endif
   ifeq ($(ARCH), riscv64)
-    BASE_URL := http://fedora.riscv.rocks/repos-dist/f$(BASE_OS_VERSION)/latest/$(ARCH)/Packages/
+#    BASE_URL := http://fedora.riscv.rocks/repos-dist/f$(BASE_OS_VERSION)/latest/$(ARCH)/Packages/
+    BASE_URL := https://secondary.fedoraproject.org/pub/alt/risc-v/archive/RPMS/riscv64
   else
     LATEST_ARCHIVED_OS_VERSION := 35
     ifeq ($(filter x86_64 armhfp, $(ARCH)), )
@@ -92,9 +93,9 @@ endif
 # Define external dependencies
 
 # Latest that could be made to work.
-GCC_VER := 13.2.0
-ifeq ($(GCC_VER), 13.2.0)
-  gcc_ver := gcc-13.2.0
+GCC_VER := 14.2.0
+ifeq ($(GCC_VER), 14.2.0)
+  gcc_ver := gcc-14.2.0
   binutils_ver := binutils-2.41
   ccache_ver := ccache-3.7.12
   mpfr_ver := mpfr-4.2.0
